pg_search/src/postgres/customscan/builders/custom_path.rs

```diff
@@ -315,6 +334,19 @@ impl<P: Into<*mut pg_sys::List> + Default> CustomPathBuilder<P> {
             .into_iter()
             .fold(0, |acc, flag| acc | flag as u32);
 
+        // Ensure presorted flag works with pathkeys
+        unsafe {
+            if (self.custom_path_node.flags & 0x0001) != 0 {
+                // CUSTOMPATH_PRESORTED flag is set
+                // If we're presorted but have no pathkeys, add a dummy one to ensure proper signaling
+                let pathkeys =
+                    PgList::<pg_sys::PathKey>::from_pg(self.custom_path_node.path.pathkeys);
+                if pathkeys.is_empty() {
+                    pgrx::warning!("presorted flag set but no pathkeys present");
+                }
```

Comment:
Is this actually an error condition (if `pathkeys.is_empty()`)?  Otherwise, it doesn't appear that a "dummy" is being added.

Reply:
This whole section also got removed. Thanks for mentioning it.